crates/kubelet/src/provider.rs

```diff
@@ -69,7 +69,10 @@ pub trait Provider {
         namespace: String,
         pod: String,
         container: String,
-    ) -> anyhow::Result<Vec<u8>>;
+        sender: hyper::body::Sender,
```

Comment:
I am not sure if there is a way around this, but it feels a little...dirty? It just feels weird to be passing a hyper specific sender all the way down the stack and making it part of a public trait. Is there any way we can get around this or make it more generic? 

Once again, @rylev your thoughts would be appreciated here

Reply:
I agree, it is not ideal, initially I tried to at least stick to using something from tokio instead. I can explore adding a new type so that `kubelet` can change the underlying implementation in the future without breaking provider implementations.

Reply:
Created a new type to wrap this. Should present a cleaner, more stable API to users.